node-core/components/interfaces.go

```diff
@@ -93,6 +93,8 @@ type (
 			ctx context.Context,
 			r *builder.RequestPayloadData,
 		) (ctypes.BuiltExecutionPayloadEnv, error)
+		// EvictPayload evicts a local payload from the cache if it exists.
+		EvictPayload(slot math.Slot, parentBlockRoot common.Root)
```

Comment:
I am experimenting in https://github.com/berachain/beacon-kit/pull/2953 whether we can drop the `parentBlockRoot` side of the key. I am positive this is possible and this would simplify the code but maybe we don't want to mix concerns here.